framework/aster-frame/src/sync/condvar.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{collections::VecDeque, sync::Arc};
+use core::{
+    sync::atomic::{AtomicBool, Ordering},
+    time::Duration,
+};
+
+use super::{Mutex, MutexGuard, SpinLock, WaitQueue};
+
+/// A condition variable.
+pub struct Condvar {
+    wait_queue: WaitQueue,
```

Comment:
The typical usage of `Condvar` for Mlsdisk is interruptible, but `WaitQueue` is not. So please implement CondVar based on `Pauser` in `aster-nix`.

Reply:
Certainly, Rust for Linux does include a [feature](https://github.com/Rust-for-Linux/linux/blob/rust-next/rust/kernel/sync/condvar.rs#L159) of this nature.

I've taken the initiative to reimplement `Condvar` using `Pauser`. However, I've encountered an issue where the current design of `Pauser` appears to be ill-suited for kernel use, resulting in a panic [here](https://github.com/asterinas/asterinas/blob/main/kernel/aster-nix/src/process/signal/pauser.rs#L115).
Consequently, it seems that this particular design of Condvar may not be compatible with kernel threads.